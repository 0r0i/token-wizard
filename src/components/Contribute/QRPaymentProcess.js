import React from 'react'
import QRCode from 'qrcode.react'
import { CopyToClipboard } from 'react-copy-to-clipboard'

const QRPaymentProcess = ({ crowdsaleProxyAddr, registryExecAddr, txData }) => {
  const targetContractStr = crowdsaleProxyAddr ? 'Proxy' : 'RegistryExec'
  const targetAddr = crowdsaleProxyAddr ? crowdsaleProxyAddr : registryExecAddr
  return (
    <div>
      <div className="payment-process">
        <div className="payment-process-qr">
          <QRCode value={targetAddr} />
        </div>
        <p className="payment-process-hash">{targetAddr}</p>

        <CopyToClipboard text={targetAddr}>
          <a href="" onClick={e => e.preventDefault()} className="payment-process-copy">
            Copy Address
          </a>
        </CopyToClipboard>

        <div className="payment-process-notation">
          <p className="payment-process-notation-title">Important</p>
          <p className="payment-process-notation-description">
            Send ethers to the Auth-os {targetContractStr} smart-contract address with a data: {txData}
          </p>
        </div>
      </div>
    </div>
  )
}

export default QRPaymentProcess
