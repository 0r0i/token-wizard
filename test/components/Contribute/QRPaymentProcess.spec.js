import React from 'react'
import render from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount } from 'enzyme'
import QRPaymentProcess from '../../../src/components/Contribute/QRPaymentProcess'

configure({ adapter: new Adapter() })

const crowdsaleProxyAddr = '0xcbf4eb5e9743c335631afe21e158bf1bb21b2864'
const registryExecAddr = '0xcbf4eb5e9743c335631afe21e158bf1bb21b2864'
const txData =
  '0x55f8650100000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000004a6f2ae3a00000000000000000000000000000000000000000000000000000000'

HTMLCanvasElement.prototype.getContext = () => {
  // return whatever getContext has to return
}

describe(`QRPaymentProcess`, () => {
  it(`should contain a canvas which draws the QR code`, () => {
    const wrapper = mount(<QRPaymentProcess crowdsaleProxyAddr={crowdsaleProxyAddr} txData={txData} />)
    const QRContainer = wrapper.find('.cnt-QRPaymentProcess_QR')
    const theCanvas = QRContainer.find('canvas')

    expect(theCanvas.exists()).toBeTruthy()
  })

  it(`should contain a copy button`, () => {
    const wrapper = mount(<QRPaymentProcess crowdsaleProxyAddr={crowdsaleProxyAddr} txData={txData} />)
    const copyButton = wrapper.find('.sw-ButtonCopyToClipboard')

    expect(copyButton.exists()).toBeTruthy()
  })

  it(`should render QRPaymentProcess component`, () => {
    // Given
    const component = render.create(<QRPaymentProcess registryExecAddr={registryExecAddr} txData={txData} />)

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })
})
