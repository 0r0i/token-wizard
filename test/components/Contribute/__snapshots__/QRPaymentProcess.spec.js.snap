// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRPaymentProcess should render QRPaymentProcess component 1`] = `
<div
  className="cnt-QRPaymentProcess"
>
  <div
    className="cnt-QRPaymentProcess_QR"
  >
    <canvas
      height={115}
      style={
        Object {
          "height": 115,
          "width": 115,
        }
      }
      width={115}
    />
  </div>
  <div
    className="cnt-QRPaymentProcess_HashContainer"
  >
    <span
      className="cnt-QRPaymentProcess_Hash"
    >
      0xcbf4eb5e9743c335631afe21e158bf1bb21b2864
    </span>
    <button
      className="sw-ButtonCopyToClipboard "
      data-clipboard-action="copy"
      data-clipboard-text="0xcbf4eb5e9743c335631afe21e158bf1bb21b2864"
      onClick={[Function]}
      type="button"
    />
  </div>
  <div
    className="cnt-QRPaymentProcess_Note"
  >
    <h3
      className="cnt-QRPaymentProcess_NoteTitle"
    >
      Important
    </h3>
    <p
      className="cnt-QRPaymentProcess_NoteDescription"
    >
      Send ethers to the Auth-os 
      RegistryExec
       smart-contract address with a MethodID: 
      0x55f8650100000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000004a6f2ae3a00000000000000000000000000000000000000000000000000000000
    </p>
  </div>
</div>
`;
